MyFridge/MyFridge/Program.cs

```diff
@@ -0,0 +1,176 @@
+            Kosher kosher = (Kosher)Enum.Parse(typeof(Kosher), Console.ReadLine() ?? "it is not kosher ");
+                foreach (var item in refrigerator.WantToEat(kosher, kind))
+                {
+                    Console.WriteLine(item.ToString());
+                }
+            }
+            catch(Exception ex)
+            {
+                Console.WriteLine(ex.Message+ "Try again - ");
+            }
+           
+
+        }
+        public static void Cleaning (Refrigerator refrigerator)
+        {
+            try
+            {
+                foreach (var item in refrigerator.SortOfExpiryDate())
+                {
+                    if (item.ExpiryDate < DateTime.Now) Console.WriteLine("thrown out:");
+                    Console.WriteLine(item.Name);
+                }
+                refrigerator.Cleaning();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message + "Try again - ");
+
+            }
+        }
+        public static void InputItem(Refrigerator refrigerator)
+        {
+            Console.WriteLine("get item: get name,kind(Food Drink),kosher(Meaty Dairy Fur ) expiryDate,space");
+            try
+            {
+                string name = Console.ReadLine() ?? "it is not null ", s = (Console.ReadLine() ?? "not null"), koshrs = Console.ReadLine() ?? "not null", expiryDates = Console.ReadLine() ?? "not null", spaces = Console.ReadLine() ?? "not null";
+                KindItem kind = (KindItem)Enum.Parse(typeof(KindItem), s);
+                Kosher kosher = (Kosher)Enum.Parse(typeof(Kosher), koshrs);
+                DateTime expiryDate = DateTime.Parse(expiryDates);
+                int space = int.Parse(spaces);
+                var inputItem = new Item(name, kind, kosher, expiryDate, space);
+                refrigerator.InsertingItem(inputItem);
+                Console.WriteLine("The food entered successfully");
+            }
+             catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message + "Try again - ");
+
+            }
+        }
+        public static void HoutItem(Refrigerator refrigerator)
+        {
+            try
+            {
+                Console.WriteLine("Give an ID number of an item to take out of the fridge");
+                int id = int.Parse(Console.ReadLine() ?? "it is not null ");
+                Console.WriteLine(refrigerator.TookItem(id));
+                Console.WriteLine("The food came out successfully");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message + "Try again - ");
+
+            }
+
+        }
+            public static void SortByAvailability(List<Refrigerator> refrigerators)
+        {
+            refrigerators.Sort((p1, p2) => p1.GetFreeSpace().CompareTo(p2.GetFreeSpace()));
+            
+        }
+        static void Main(string[] args)
```

Comment:
Note the program is crashing. The most important thing when submitting a task is:
1 - it should compile
2 - it should run
3 - the logic should be correct
even at the cost of not completing all sub-tasks.

please make it run :)

Reply:
Sorry, I probably didn't save recent changes before I uploaded to github, the program ran fine.